### PR TITLE
🐛 send multiple messages if solution is too big

### LINKED ISSUE #959: produce some kind of result when the solution factory is large
Discord embeds only support 25 fields, and currently 1 field is used per recipe. For larger factories, the command just crashes once it goes to send the message. The solver itself finds a solution just fine. 

Seeing large factories is still useful for planning, so maybe a basic text output for these is better than nothing.

### PR DESCRIPTION
**Pull request by twentylemon**
##### Summary

Fixes #959

Discord has a message size limit, which sometimes big solutions will run into. This chunks up the message into the biggest possible components that can fit into a single message, then send multiple messages.

##### Checklist

- [x] this is a source code change
  - [x] run `format` in the repository root
  - [x] run `pytest` in the repository root
  - [x] link to issue being fixed using a [_fixes keyword_](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue), if such an issue exists
  - [ ] update the wiki documentation if necessary
- [ ] or, this is **not** a source code change

### CODE REVIEW

File: duckbot/util/embeds/__init__.py [hunk 1]
```diff
@@ -7,3 +11,18 @@
...
+    total, group, groups = 0, [], []
+    for embed in embeds:
+        length = len(embed)
+        if total + length < MAX_EMBED_LENGTH:
+            total += length
+            group.append(embed)
+        else:
+            total = 0
+            groups.append(group)
+    return groups + [group] if group else groups
```
Comment (jameshughes89) [inline duckbot/util/embeds/__init__.py:28]: jesus

File: tests/util/embeds/group_by_max_length_test.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+from discord import Embed
+
+from duckbot.util.embeds import group_by_max_length
+
+
+def embed(length: int) -> Embed:
+    return Embed(title="a" * length)
+
+
+@pytest.mark.parametrize("length", [0, 1, 3000, 6001])
```
Comment (jameshughes89) [inline tests/util/embeds/group_by_max_length_test.py:11]: are 3000 and 6001 meaningful?

Comment (jameshughes89) [reply tests/util/embeds/group_by_max_length_test.py:11]: https://github.com/duck-dynasty/duckbot/pull/1006/files#diff-852d46fbcd57025c42fb5af4c1d67bca90bcaf8ad58ab1362b17cbda69194db3L2

OK, I see

Comment (twentylemon) [reply tests/util/embeds/group_by_max_length_test.py:11]: 6000 is the max embed length discord has. 3000 is just a non-1 number within the size limit

**Review by jameshughes89 — APPROVED** [review]

File: duckbot/util/embeds/__init__.py (hunk 1 above)
Comment (twentylemon) [reply duckbot/util/embeds/__init__.py:28]: I imagine there's a functional way to do this, but I can't think of it right now.

**Pull request merged**